modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/strategy/NestedFieldsSupportMapToDDMFormValuesConverterStrategy.java

```diff
@@ -76,13 +76,17 @@ private void _addMissingValues(
 		for (Map.Entry<String, DDMFormField> entry :
 				ddmFormFieldsMap.entrySet()) {
 
-			Set<String> keys = values.keySet();
+			List<String> keys = TransformUtil.transform(
```

Comment:
@JanisZhang Please use a for-loop here, it will not create any list/array which will soon become garbage after the isEmpty() check. For runtime code we prefer performance.